### PR TITLE
2808 convert to es6 classes

### PR DESCRIPTION
**Pull request by eleanor-heath**
https://github.com/adaptlearning/adapt_framework/issues/2808

### CODE REVIEW

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

File: js/componentNameView.js [hunk 1]
```diff
@@ -1,83 +1,77 @@
...
+  /**
+   * Used to enable the question during submit and complete stages
+   */
+  enableQuestion() {}
+
+  /**
+   * Used to reset the question when revisiting the component
+   * @param {string} [type] - the type of reset e.g. hard/soft
+   */
+  resetQuestionOnRevisit(type) {}
```
Comment (oliverfoster) [inline js/componentNameView.js]: This has been removed and is instead implemented in the model at `.reset()`, the function in the model should be:
```js
rest(type = 'hard', canReset = this.get('_canReset')) {
  super.reset(type, canReset);
}
```

File: js/componentNameView.js [hunk 2]
```diff
@@ -1,83 +1,77 @@
...
+   */
+  getResponse() {}
+
+  /**
+   * Used by adapt-contrib-spoor to get the type of this question in the
+   * format required by the cmi.interactions.n.type data field.
+   * Please note the answer will not store correctly unless this function returns a valid string
+   * @returns {string} one of the following: choice, matching, numeric, fill-in
+   */
+  getResponseType() {}
```
Comment (oliverfoster) [inline js/componentNameView.js]: These should be in the model.

File: templates/componentName.hbs [hunk 3]
```diff
@@ -1,7 +1,7 @@
-<div class="componentName-inner component-inner" role="region" aria-label="{{_globals._components._componentName.ariaRegion}}">
+<div class="componentName__inner component__inner" role="region" aria-label="{{_globals._components._componentName.ariaRegion}}">
```
Comment (oliverfoster) [inline templates/componentName.hbs]: This should be jsx

**Review by oliverfoster — CHANGES_REQUESTED** [review]
There are a few other bits I need to check. So this is just a preliminary review.

File: bower.json [hunk 4]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "adapt-componentName",
   "description": "DESCRIPTION",
-  "version": "0.0.1",
-  "framework": ">=2.2.0",
+  "version": "1.0.0",
+  "framework": ">=5.8.0",
```
Comment (oliverfoster) [inline bower.json]: Should be v5.14 for jsx fixes https://github.com/adaptlearning/adapt_framework/releases/tag/v5.14.0

File: js/componentNameModel.js [hunk 5]
```diff
@@ -1,72 +1,86 @@
...
 
-            this.set("_userAnswer", userAnswer);
-        },
+  /**
+   * This evaluates the user's answer and stores the value
+   * This value can then be used later on e.g. by the view to show the user's answer
+   */
+  storeUserAnswer() {
+    // Expand on this to retrieve the user's answer as a single value
+    const userAnswer;
```
Comment (oliverfoster) [inline js/componentNameModel.js]: Comment suggestion:
```suggestion
    // Note: _userAnswer can only contain arrays, booleans and numbers. 
    // Each array must contain the same type, such that only arrays of arrays, 
    // arrays of booleans and arrays of numbers are allowed. 
    // [[true, false], [0, 1, 2, 3]]
    // [0, 1, 2, 3]
    // [true, false]
    const userAnswer;
```

File: js/componentNameModel.js [hunk 6]
```diff
@@ -1,72 +1,86 @@
...
+    if (!this.get('_isSubmitted')) return;
 
-            this.setQuestionAsSubmitted();
-            this.markQuestion();
-            this.setScore();
-            this.setupFeedback();
-        },
+    // The user answer is retrieved here
+    // This value can then be used to set individual answers on items
+    const userAnswer = this.get('_userAnswer');
```
Comment (oliverfoster) [inline js/componentNameModel.js:27]: Comment suggestion:
```suggestion
    // Note: _userAnswer can only contain arrays, booleans and numbers. 
    // Each array must contain the same type, such that only arrays of arrays, 
    // arrays of booleans and arrays of numbers are allowed. 
    // [[true, false], [0, 1, 2, 3]]
    // [0, 1, 2, 3]
    // [true, false]
    const userAnswer = this.get('_userAnswer');
```

File: js/componentNameView.js [hunk 7]
```diff
@@ -1,83 +1,56 @@
...
+
+  /**
+   * Used by question components instead of postRender
+   */
+
+  onQuestionRendered() {}
+
+  /**
+   * Called when the user clicks submit and this.model.canSubmit() returns false
+   * Not necessary but you
```
Comment (oliverfoster) [inline js/componentNameView.js]: Comment unfinished

File: js/componentNameView.js [hunk 8]
```diff
@@ -1,83 +1,56 @@
...
+   */
+  onCannotSubmit() {}
+
+  /**
+   * This is important and should give the user visual feedback on how they answered the question
+   * Normally done through ticks and crosses by adding classes
+   */
+  showMarking() {}
```
Comment (oliverfoster) [inline js/componentNameView.js]: With jsx this isn't necessary as [`isInteractive() && _canShowMarking`](https://github.com/adaptlearning/adapt-contrib-core/blob/master/js/models/questionModel.js#L282) determine the show/hide of marking in the template.

[MCQ example](https://github.com/adaptlearning/adapt-contrib-mcq/blob/master/templates/mcq.jsx#L26)

Perhaps we should make that into a model property called `shouldShowMarking`? Woud make sense to have that as core behaviour.

Or at least include the mcq example in the template here.

What do you think? Can you create an issue for adding `shouldShowMarking` if you agree?

File: js/componentNameView.js [hunk 9]
```diff
@@ -1,83 +1,56 @@
...
+   */
+  showCorrectAnswer() {}
+
+  /**
+   * Displays the user's answer and hides the correct answer
+   * Should use the values stored in this.storeUserAnswer
+   */
+  hideCorrectAnswer() {}
+
+}
```
Comment (oliverfoster) [inline js/componentNameView.js]: These three functions aren't necessary with jsx.

Although I think `resetQuestion` has to stay for the moment as it's [programmically necessary](https://github.com/adaptlearning/adapt-contrib-core/blob/master/js/views/questionView.js#L291) as can be seen in the [mcq view](https://github.com/adaptlearning/adapt-contrib-mcq/blob/master/js/McqView.js#L61-L64). It should really be just a model function.

Can you create a ticket to make `resetQuestion` a model function?

**Review by oliverfoster — CHANGES_REQUESTED** [review]

File: js/componentNameView.js [hunk 10]
```diff
@@ -1,83 +1,56 @@
...
+
+  /**
+   * Used to disable the question during submit and complete stages
+   */
+  disableQuestion() {}
+
+  /**
+   * Used to enable the question during submit and complete stages
+   */
+  enableQuestion() {}
```
Comment (oliverfoster) [inline js/componentNameView.js]: There two functions aren't necessary with jsx

File: js/componentNameView.js (hunk 7 above)
Comment (eleanor-heath) [reply js/componentNameView.js]: I have no idea what this is should say 🤣

Comment (eleanor-heath) [reply js/componentNameView.js]: I may just remove it

File: js/componentNameView.js (hunk 9 above)
Comment (eleanor-heath) [reply js/componentNameView.js]: I have added an issue to make resetQuestion a model here https://github.com/adaptlearning/adapt-contrib-core/issues/31.

File: js/componentNameView.js (hunk 8 above)
Comment (eleanor-heath) [reply js/componentNameView.js]: I have created a ticket to add the model property `shouldShowMarking` here https://github.com/adaptlearning/adapt-contrib-core/issues/32

**Review by ethan-lp — APPROVED** [review]

**Comment by oliverfoster** [comment]
do not merge, pending further changes after https://github.com/adaptlearning/adapt-contrib-core/pull/39 https://github.com/adaptlearning/adapt-contrib-core/pull/38

**Review by chris-steele — APPROVED** [review]

**Comment by oliverfoster** [comment]
need to wait for the core prs to be released and then we can update this to match
 https://github.com/adaptlearning/adapt-contrib-core/pull/39 
https://github.com/adaptlearning/adapt-contrib-core/pull/38

Awaiting release of adapt_framework for adapt-contrib-core v6.1.0

**Review by lc-alexanderbenesch — APPROVED** [review]
👀

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Comment by oliverfoster** [comment]
@eleanor-heath could you review the changes I've made and let me know?

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Comment by eleanor-heath** [comment]
@oliverfoster looks good to me 👍

**Review by lc-alexanderbenesch — APPROVED** [review]
looks good 👍

**Comment by oliverfoster** [comment]
Awaiting https://github.com/adaptlearning/adapt-contrib-core/pull/72 merge and release. fw version needs bumping accordingly thereafter. Needs deprecated `setScore` behaviour.

**Pull request merged**